src/openrct2/core/String.cpp

```diff
@@ -343,65 +333,60 @@ namespace String
         return buffer;
     }
 
-    utf8* Format(const utf8* format, ...)
+    std::string Format(const utf8* format, ...)
     {
         va_list args;
         va_start(args, format);
-        utf8* result = Format_VA(format, args);
+        std::string result = Format_VA(format, args);
         va_end(args);
         return result;
     }
 
-    utf8* Format_VA(const utf8* format, va_list args)
+    std::string Format_VA(const utf8* format, va_list args)
     {
         va_list args1, args2;
         va_copy(args1, args);
         va_copy(args2, args);
 
         // Try to format to a initial buffer, enlarge if not big enough
         size_t bufferSize = 4096;
-        utf8* buffer = Memory::Allocate<utf8>(bufferSize);
+        std::string buffer;
+        buffer.resize(bufferSize - 1);
 
-        // Start with initial buffer
-        int32_t len = vsnprintf(buffer, bufferSize, format, args);
+        int32_t len = vsnprintf(buffer.data(), 0, format, args);
```

Comment:
You've removed the bufferSize this is causing the tests to fail.

Reply:
A buffer size of 0 is alright to get the humber of byte sit would write, excluding the null terminator. The buffer itself doesn't have to be passed, it can simple be nullptr here.

```suggestion
        int32_t len = vsnprintf(nullptr, 0, format, args);
```

Reply:
Sure you could do that resize the string based on the result followed by another call that actually does the work. But at present that is not what is being done so the easiest thing would be to just set the buffer size.